org.sonarlint.eclipse.core/src/org/sonarlint/eclipse/core/internal/jobs/AnalyzeProjectJob.java

```diff
@@ -119,24 +132,85 @@ public void deleteUnmatched() throws CoreException {
   }
 
   @Override
-  protected IStatus run(SonarRunnerFacade runner, final IProgressMonitor monitor) {
-
-    // Configure
-    Properties properties = new Properties();
-    Collection<ProjectConfigurator> usedConfigurators = SonarLintAnalysisConfigurator.configureAnalysis(request, properties, extraProps, monitor);
+  protected IStatus run(SonarLintClientFacade sonarlint, final IProgressMonitor monitor) {
 
     // Analyze
     Collection<File> tmpToDelete = new ArrayList<>();
     try {
+      // Configure
+      IProject project = request.getProject();
+      IPath projectSpecificWorkDir = project.getWorkingLocation(SonarLintCorePlugin.PLUGIN_ID);
+      Map<String, String> mergedExtraProps = new LinkedHashMap<>();
+      final List<IFile> filesToAnalyze;
       if (request.getOnlyOnFiles() != null) {
-        IProject project = request.getProject();
-        final File baseDir = project.getLocation().toFile();
-        handleLinkedFiles(tmpToDelete, baseDir);
+        filesToAnalyze = new ArrayList<>(request.getOnlyOnFiles().size());
+        filesToAnalyze.addAll(request.getOnlyOnFiles());
+      } else {
+        filesToAnalyze = new ArrayList<>();
+        project.accept(new IResourceProxyVisitor() {
+
+          @Override
+          public boolean visit(IResourceProxy proxy) throws CoreException {
+            if (proxy.getType() == IResource.FILE) {
+              filesToAnalyze.add((IFile) proxy.requestResource());
+              return false;
+            }
+            return true;
+          }
+        }, IResource.DEPTH_INFINITE, IContainer.EXCLUDE_DERIVED);
+      }
+
+      Collection<ProjectConfigurator> usedConfigurators = configure(project, filesToAnalyze, mergedExtraProps, monitor);
+
+      handleLinkedFiles(tmpToDelete, filesToAnalyze);
+
+      List<InputFile> inputFiles = new ArrayList<>(filesToAnalyze.size());
+      String allTestPattern = PreferencesUtils.getTestFileRegexps();
+      String[] testPatterns = allTestPattern.split(",");
+      final List<PathMatcher> pathMatchersForTests = new ArrayList<>();
+      FileSystem fs = FileSystems.getDefault();
+      for (String testPattern : testPatterns) {
+        pathMatchersForTests.add(fs.getPathMatcher("glob:" + testPattern));
+      }
+      for (final IFile file : filesToAnalyze) {
+        final java.nio.file.Path filePath = file.getRawLocation().makeAbsolute().toFile().toPath();
+        inputFiles.add(new InputFile() {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Reduce this anonymous class number of lines from 26 to at most 20, or make it a named class. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS1188)